### PR TITLE
Update chatgpt-api to use proper api

### LINKED ISSUE #75: Switch to use the official API
It is now possible to access the ChatGPT model via the official API. Right now this is undocumented and unreleased.

Changes are minimal from a code point of view but from a users point of view they will have to add an API key.

We will therefore bump the major version number.

### PR DESCRIPTION
**Pull request by max298**
closes #75 

This change updates the transitive-bullshit/chatgpt-api to version 4.0.4 which now [uses the official API](https://github.com/transitive-bullshit/chatgpt-api/commit/21dd9d518f106eecbc5102346c2f37d3bd68a761).

### CODE REVIEW

**Comment by bertybuttface** [comment]
Looks good, one slight comment is the `yarn.lock` but thats not really on you, this package already got that wrong with `package-lock.json`. Lets clear that up on another PR.

**Comment by max298** [comment]
I just noticed this will also close #72 as puppeteer is no longer needed and as well close #76 😀

**Comment by max298** [comment]
The authors from the Node.js wrapper around ChatGPT just set the model to the 2022 version (see (here)[https://github.com/transitive-bullshit/chatgpt-api/compare/v4.1.0...v4.1.1]). I've bumped the version to 4.1.1 am currently running the bot again without any issues. However I cant decide how the maintainers feel about this, as we might end up updating the package almost daily as nobody knows for how long the current model will last...

File: src/index.ts [hunk 1]
```diff
@@ -42,18 +42,10 @@ async function main() {
...
 
   // use puppeteer to bypass cloudflare (headful because of captchas)  
-  const chatGPT: ChatGPTAPIBrowser = new ChatGPTAPIBrowser({
-    email: OPENAI_EMAIL,
-    password: OPENAI_PASSWORD,
-    isGoogleLogin: (OPENAI_LOGIN_TYPE == "google"), 
-    isMicrosoftLogin: (OPENAI_LOGIN_TYPE == "microsoft"),
-    isProAccount: OPENAI_PRO
+  const chatGPT: ChatGPTAPI = new ChatGPTAPI({
+	  apiKey: OPENAI_API_KEY
```
Comment (max298) [inline src/index.ts:46]: Another idea could be to introduce a new environment variable `OPENAI_MODEL` with a default of `text-chat-davinci-002-20221122` (as of today) so the users can change to model on their own as [shown here](https://github.com/transitive-bullshit/chatgpt-api/issues/292#issuecomment-1414304348).

Comment (bertybuttface) [reply src/index.ts:46]: This sounds like the correct direction to go in. Thanks for all your work tracking these changes.

Comment (max298) [reply src/index.ts]: Sure, I really enjoy talking to ChatGPT directly using my matrix client, so thank you for your working creating this bot.

I've just added a environment variable `CHATGPT_MODEL` as suggested

**Pull request merged**